backend/models/core/Resource.js

```diff
@@ -0,0 +1,154 @@
+import DB from "@backend/data_access/DB";
+import {
+  stringTypes,
+  longStringTypes,
+  longValueTypes,
+} from "@/components/cmsTypes";
+
+class Resource {
+  static async get({ slug }) {
+    const db = new DB();
+    const getResourceSQL = `SELECT * FROM \`resources\` WHERE slug = :slug`;
+
+    const resource = await db.executeStatement(getResourceSQL, [
+      { name: "slug", value: { stringValue: slug } },
+    ]);
+
+    return resource.records.map(
+      ([
+        { longValue: id },
+        { stringValue: slug },
+        { stringValue: type },
+        { stringValue: value_string },
+        { stringValue: value_long_string},
+        { longValue: value_double}
+      ]) => ({
+        id,
+        slug,
+        type,
+        value_string,
+        value_long_string,
+        value_double
+      })
+    );
+  }
+
+  static async create({ name, key, type }) {
+    const db = new DB();
+    const getResourceSQL = `SELECT * FROM \`resources\` WHERE slug = :slug`;
+
+    const resource = await db.executeStatement(getResourceSQL, [
+      { name: "slug", value: { stringValue: name } },
+    ]);
+  
+    if (!resource.records.length) {
+ 
+      const insertResourceSQL =
+        "INSERT into `resources` (`slug`, `type`, `value_string`, `value_long_string`, `value_double`) VALUES (:slug, :type, :value_string, :value_long_string, :value_double)";
```

Comment:
* Props to using `resources` as a polymorphic, multi-type model, (similar to `values`).
* Although, in this case, would it be simpler if we serialize all types to string?
* Another strategy would be to store everything as string, then let the logic code decide on the data type.
* If we simplify this and store everything as string, it would be like a key-value store, similar to Redis and Memcache.